lib/canonicalization.ts

```diff
@@ -48,21 +49,27 @@ const createBidMap = (graph: Quads): ReadonlyMap<rdf.BlankNode, BNodeId> => {
  * 
  * @param state - the overall canonicalization state + interface to the underlying RDF environment
  * @param input
- * @param deduplicate - whether duplicate quads should be removed from the input
- * @returns - A semantically identical set of Quads, with canonical BNode labels, plus other information.
+ * @param copy - whether the input should be copied to a local store (e.g., if the input is a generator, or the uniqueness of quads are not guaranteed). If this
+ * parameter is not used (i.e., value is `undefined`) the copy is always done _unless_ the input is an `rdf.DatasetCore` instance.
+ * @returns - A semantically identical set of Quads using canonical BNode labels, plus other information.
  * 
  * @async
  */
-export async function computeCanonicalDataset(state: GlobalState, input: InputDataset, deduplicate = false): Promise<C14nResult> {
+export async function computeCanonicalDataset(state: GlobalState, input: InputDataset, copy: boolean | undefined = undefined): Promise<C14nResult> {
     const finalInput = async (): Promise<InputQuads> => {
         if (typeof input === 'string') {
             return await parseNquads(input as string);
-        } else if (deduplicate) {
-            const retval = new n3.Store();
-            for (const quad of input) retval.add(quad);
-            return retval;
         } else {
-            return input;;
+            const createCopy = (): InputQuads => {
+                const retval = new n3.Store();
+                for (const quad of input) retval.add(quad);
+                return retval;
+            };
+            if (copy === undefined) {
+                return isQuads(input) ? input : createCopy();
+            } else {
+                return copy ? createCopy() : input;
+            }
```

Comment:
```suggestion
            if (copy ?? !isQuads(input)) {
                const retval = new n3.Store();
                for (const quad of input) retval.add(quad);
                return retval;
            }  
            return input;
```

Note that `??` is the Nullish coalescing operator https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing